src/core/execute.c

```diff
@@ -3475,6 +3481,28 @@ static int compile_bind_mounts(
                 }
         }
 
+        if (context->protect_control_groups && memory_pressure_path && !streq(memory_pressure_path, "/dev/null")) {
+                _cleanup_free_ char *s = NULL, *d = NULL;
+
+                s = strdup(memory_pressure_path);
+                if (!s) {
+                        r = -ENOMEM;
+                        goto finish;
+                }
+
+                d = strdup(memory_pressure_path);
+                if (!d) {
+                        r = -ENOMEM;
+                        goto finish;
+                }
+
+                bind_mounts[h++] = (BindMount) {
+                        .source = TAKE_PTR(s),
+                        .destination = TAKE_PTR(d),
+                        .read_only = false,
+                };
+        }
+
```

Comment:
this looks like the wrong place to add this. instead, please create copy of context->read_write_paths and merge it into that, and pass as fifth argument to setup_namespace() below.

this should not be implemented via the `BindMount` struct, since this will be weird in RootImage= + RootDirectoy= scenarios...

Reply:
ok